File: Leetcode/string/680_ValidPalindromeII/Solution.py
class Solution(object):
    def validPalindrome(self, s):
        """
        :type s: str
        :rtype: bool
        """
        start, end = 0, len(s)-1
        while start<end:
            if s[start]!=s[end]:
                slice1 = s[start:end]
                slice2 = s[start+1:end+1]
                return slice1==slice1[::-1] or slice2==slice2[::-1]
            start+=1; end-=1
        return True

File: Leetcode/string/680_ValidPalindromeII/Solution.java
class Solution {
    public boolean validPalindrome(String s) {
        int start = 0, end = s.length()-1;
        while(start<end){
            if(s.charAt(start)!=s.charAt(end))
                return isPalindrome(s.substring(start, end))||isPalindrome(s.substring(start+1, end+1));
            start++; end--;
        }
        return true;
    }
    private boolean isPalindrome(String s){
        int start = 0, end = s.length()-1;
        while(start<end){
            if(s.charAt(start)!=s.charAt(end))
                return false;
            start++; end--;
        }
        return (s.length()==0)? false: true;
    }
}
